pages/chaoss/visualizations/contrib_activity_cycle.py

```diff
@@ -0,0 +1,201 @@
+    cache = cm()
+    df = cache.grabm(func=cmq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cmq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, interval)
+
+    fig = create_figure(df, interval)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, interval):
+
+    # for this usecase we want the datetimes to be in their local values
+    # tricking pandas to keep local values when UTC conversion is required for to_datetime
+    df["author_timestamp"] = df["author_timestamp"].astype("str").str[:-6]
+    df["committer_timestamp"] = df["committer_timestamp"].astype("str").str[:-6]
+
+    # convert to datetime objects rather than strings
+    df["author_timestamp"] = pd.to_datetime(df["author_timestamp"], utc=True)
+    df["committer_timestamp"] = pd.to_datetime(df["committer_timestamp"], utc=True)
+    df.loc[df["author_timestamp"] == df["committer_timestamp"], "author_timestamp"] = None
+
+    df_final = pd.DataFrame()
+
+    if interval == "H":
+        hour = pd.concat([df["author_timestamp"].dt.hour, df["committer_timestamp"].dt.hour])
+        df_hour = pd.DataFrame(hour, columns=["Hour"])
+        df_final = df_hour.groupby(["Hour"])["Hour"].count()
+    else:
+        weekday = pd.concat([df["author_timestamp"].dt.day_name(), df["committer_timestamp"].dt.day_name()])
+        df_weekday = pd.DataFrame(weekday, columns=["Weekday"])
+        df_final = df_weekday.groupby(["Weekday"])["Weekday"].count()
+
+    # combine the weekday values for author and committer
+    """weekday = pd.concat([df["author_timestamp"].dt.day_name(), df["committer_timestamp"].dt.day_name()])
+    hour = pd.concat([df["author_timestamp"].dt.hour, df["committer_timestamp"].dt.hour])
+    df_final = pd.DataFrame(weekday, columns=["Weekday"])
+    df_final["Hour"] = hour"""
+
+    return df_final
+
+
+def create_figure(df: pd.DataFrame, interval):
+
+    column = "Weekday"
+    order = ["Saturday", "Friday", "Thursday", "Wednesday", "Tuesday", "Monday", "Sunday"]
+    if interval == "H":
+        column = "Hour"
+
+    """fig = px.histogram(df, x=column, color_discrete_sequence=[color_seq[3]])
+    if interval == "D":
+        fig.update_xaxes(
+            categoryorder="array",
+            categoryarray=["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"],
+        )
+    fig.update_layout(
+        yaxis_title="Activity Count",
+        font=dict(size=14),
+    )"""
+
+    fig = px.bar(df, y=column, color_discrete_sequence=[color_seq[3]])
+    fig.update_traces(hovertemplate="%{x} Activity Count: %{y}<br>")
```

Comment:
I think the 'x' value could be templated as "Hour: {x}:00"; makes more sense than having the raw hour without context in the hover.